applications/ray/kuberay-tpu-webhook/main.go

```diff
@@ -251,85 +284,72 @@ func validateRayCluster(admissionReview *admissionv1.AdmissionReview) (*admissio
 	return admissionResponse, nil
 }
 
-// add TPU_WORKER_HOSTNAMES to containers in a ray cluster
-func mutateRayCluster(admissionReview *admissionv1.AdmissionReview) (*admissionv1.AdmissionResponse, error) {
-	raycluster, err := extractRayCluster(admissionReview)
-	if err != nil {
-		return nil, err
-	}
-
-	clusterName := raycluster.Name
-	headlessServiceName = fmt.Sprintf("%s-%s", clusterName, headlessServiceSuffix)
-	var patches []patch
-	workerGroupSpecs := raycluster.Spec.WorkerGroupSpecs
-	if workerGroupSpecs == nil {
-		return nil, errors.New("WorkerGroupSpecs not specified")
-	}
-	for i := 0; i < len(workerGroupSpecs); i++ {
-		workerGroupSpec := workerGroupSpecs[i]
-		// reset past sliceToWorkers entries for ray cluster
-		groupName := workerGroupSpec.GroupName
-		podSlice := slice{clusterName, groupName}
-		sliceToWorkers[podSlice] = 0
-
-		containers := workerGroupSpec.Template.Spec.Containers
-		if containers == nil {
-			return nil, errors.New("Container path not specified")
-		}
-		if containerRequestingTPUs(containers...) {
-			topology := workerGroupSpec.Template.Spec.NodeSelector["cloud.google.com/gke-tpu-topology"]
-			isMultiHost, err := isTPUMultiHost(topology)
-			if err != nil {
-				return nil, err
-			}
-			if isMultiHost {
-				joinedHostNames, err := genDNSHostnames(workerGroupSpec)
-				if err != nil {
-					return nil, err
-				}
-				injectHostnames(joinedHostNames, workerGroupSpec, i, &patches)
+func getEnvironmentVariable(varName string, container corev1.Container) string {
+	if container.Env != nil && len(container.Env) > 0 {
+		for _, envVar := range container.Env {
+			if envVar.Name == varName {
+				return envVar.Value
 			}
 		}
 	}
-
-	patchBytes, err := json.Marshal(patches)
-	if err != nil {
-		return nil, err
-	}
-
-	// Create AdmissionResponse
-	admissionResponse := &admissionv1.AdmissionResponse{
-		UID:     admissionReview.Request.UID,
-		Allowed: true,
-		Patch:   patchBytes,
-		PatchType: func() *admissionv1.PatchType {
-			pt := admissionv1.PatchTypeJSONPatch
-			return &pt
-		}(),
-	}
-	return admissionResponse, nil
+	return ""
 }
 
-func hasWorkerID(container corev1.Container) bool {
-	if container.Env != nil && len(container.Env) > 0 {
-		for _, envVar := range container.Env {
-			if envVar.Name == "TPU_WORKER_ID" {
-				return true
+// get next replica ID to assign a pod to
+func getReplicaIndex() int {
+	if sliceToWorkers == nil {
+		return 0
+	}
+	next_lowest_id := math.MaxInt32
+	for slice, workerList := range sliceToWorkers {
```

Comment:
Looks like sliceToWorkers could contain slice information from multiple Ray clusters, would that cause any issues here?

Reply:
That's a good catch, I added in a check to make sure that it's only counting the pods/slices in the desired cluster.